src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -77,8 +101,24 @@ impl CommandRunner {
       );
     }
 
-    // Validate any configured static headers.
-    call_option(&headers, None)?;
+    let address_with_scheme = format!("{}://{}", scheme, action_cache_address);
+
+    let endpoint = Self::create_tonic_endpoint(&address_with_scheme, tls_client_config.as_ref())?;
+    let channel = tonic::transport::Channel::balance_list(vec![endpoint].into_iter());
+    let action_cache_client = Arc::new(if headers.is_empty() {
+      ActionCacheClient::new(channel)
+    } else {
+      let headers = headers.clone();
+      ActionCacheClient::with_interceptor(channel, move |mut req: Request<()>| {
+        let metadata = req.metadata_mut();
+        for (key, value) in &headers {
+          let key_ascii = AsciiMetadataKey::from_str(key.as_str()).unwrap();
```

Comment:
Ditto unwrap

Reply:
Fixed and also switched to using a single interceptor.